### ISSUE TITLE
Clang warn when building go project

### ISSUE THREAD

**Issue by ltagliamonte-dd** [pganalyze/pg_query_go#112]
Hello folks,
I'm getting this warning when I build and run unit tests in my project that uses this library:
```
# github.com/pganalyze/pg_query_go/parser
src_backend_nodes_nodeFuncs.c:907:17: warning: passing arguments to a function without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
```

**clang:**
```
clang --version
Apple clang version 15.0.0 (clang-1500.3.9.4)
Target: arm64-apple-darwin23.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by ltagliamonte-dd** [comment]
fixed updating to latest version
